src/main/java/TaskList.java

```diff
@@ -0,0 +1,232 @@
+import duke.*;
+
+public class TaskList {
+    //private Storage storage;
+
+    public TaskList() {
+        //storage = new Storage();
+    }
+    public static void printLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    //check if task description is blank (for to-do)
+    public static boolean isDescriptionBlank(String description) {
+        boolean isBlank = true;
+        for (int i = 0; i < description.length(); i++) {
+            char c = description.charAt(i);
+            isBlank = Character.isWhitespace(c); //true if character is whitespace
+        }
+        return isBlank;
+    }
+
+    //list tasks
+    public static void listTasks() {
+        printLine();
+        System.out.println("Here are the tasks in your list:");
+        int i = 1;
+        int numOfTasks = 0;
+        for (Task t : Task.tasks) {
+            System.out.println(i + "." + t);
+            i++;
+            numOfTasks++;
+        }
+        if (numOfTasks == 0) {
+            System.out.println("Looks like your list is still empty");
+        }
+        printLine();
+    }
+
+    //find task
+    public static void findTask(String[] input) {
+        try {
+            printLine();
+            String keyword = input[1];
+            System.out.println("Here are the matching tasks in your list:");
+            int i = 1;
+            int numOfMatches = 0;
+            for (Task t : Task.tasks) {
+                String s = "" + t;
+                if (s.contains(keyword)) {
+                    System.out.println(i + "." + t);
+                    i++;
+                    numOfMatches++;
+                }
+            }
+            if (numOfMatches == 0) {
+                System.out.println("I couldn't find any matching tasks :(");
+            }
+            printLine();
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("☹ OOPS!!! Keyword is missing. Please try again by giving me a keyword to look for.");
+            printLine();
+        }
+    }
+
+    //mark task as done
```

Comment:
I see a lot of code duplication in this class. You can see how you can utilize the common bits of code and put it in a function to call when necessary. [See here](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#minimise-code-duplication)